src/settings/SettingsStore.ts

```diff
@@ -438,6 +438,7 @@ export default class SettingsStore {
             settingName = setting.invertedSettingName;
             value = !value;
         }
+        if (!roomId) return;
```

Comment:
shouldn't it just not be optional?

Reply:
It should, but half of the usages have `roomId` as optional, so we need to change the API here. The question is: can we safely rework this API without deprecation, or do we need to properly deprecate it?